localstack/services/apigateway/apigateway_listener.py

```diff
@@ -121,6 +126,21 @@ def extract_path_params(path, extracted_path):
     return path_params
 
 
+def extract_query_string_params(path):
+    parsed_path = urlparse(path)
+    path = parsed_path.path
+    parsed_query_string_params = parse_qs(parsed_path.query)
+
+    query_string_params = {}
+    for query_param_name, query_param_values in parsed_query_string_params.items():
+        if len(query_param_values) <= 1:
```

Comment:
Should be changed to:
```suggestion
        if len(query_param_values) == 1:
```
(to avoid `IndexError` if the list is empty)

Reply:
oh! good catch, thanks!